### ISSUE TITLE
Dependent dates

### ISSUE THREAD

**Issue by orsinium** [olebedev/when#17]
`when` cannot parse dates with relative delta from some date:

```go
gore> r, err := w.Parse("завтра в 12:00 поезд, напомни за час", time.Now())
&when.Result{
  Index:  0,
  Text:   "завтра в 12:00",
  Source: "завтра в 12:00 поезд, напомни за час",
  Time:   2019-02-28 12:00:28 Local,
}
nil
```

BTW, I have old [rutimeparser](https://github.com/orsinium/rutimeparser) lib for Python. Maybe, some [test cases](https://github.com/orsinium/rutimeparser/blob/master/tests.py#L43) will be useful for you.

**Comment by olebedev** [comment]
PRs are very welcome :D

**Comment by orsinium** [comment]
Yeah, sure. I'm going to have a look what I can improve. At this moment I've just marked some issues to not forget to fix. You, me, someone else - it doesn't matter :)
